### ISSUE TITLE
vcpkg_fixup_cmake_targets don't fix *d.dll file path in OpenCVModules-debug.cmake

### ISSUE THREAD

**Issue by LilyWangL** [microsoft/vcpkg#8655]
**Describe the bug**
vcpkg_fixup_cmake_targets fix files *.lib path in **OpenCVModules-debug.cmake**, the files *d.dll path are wrong. See this in **OpenCVModules-debug.cmake**:
```
set_target_properties(opencv_core PROPERTIES
  IMPORTED_IMPLIB_DEBUG "${_IMPORT_PREFIX}/debug/lib/opencv_cored.lib"
  IMPORTED_LOCATION_DEBUG "${_IMPORT_PREFIX}/x86/vc16/bin/opencv_cored.dll"
  )

list(APPEND _IMPORT_CHECK_TARGETS opencv_core )
list(APPEND _IMPORT_CHECK_FILES_FOR_opencv_core "${_IMPORT_PREFIX}/debug/lib/opencv_cored.lib" "${_IMPORT_PREFIX}/x86/vc16/bin/opencv_cored.dll" )
```
Some ports depend on opencv4 will build failed due to can't find *d.dll.

**Environment**
- OS: [Windows 10]
- Compiler:  Visual Studio 2019

**To Reproduce**
1. ./vcpkg install dbow2
2. See error
```
CMake Error at F:/Lily/issue8393/vcpkg/installed/x86-windows/share/opencv/OpenCVModules.cmake:164 (message):
  The imported target "opencv_core" references the file

     "F:/Lily/issue8393/vcpkg/installed/x86-windows/x86/vc16/bin/opencv_cored.dll"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "F:/Lily/issue8393/vcpkg/installed/x86-windows/share/opencv/OpenCVModules.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  F:/Lily/issue8393/vcpkg/installed/x86-windows/share/opencv/OpenCVConfig.cmake:126 (include)
  F:/Lily/issue8393/vcpkg/scripts/buildsystems/vcpkg.cmake:263 (_find_package)
  CMakeLists.txt:36 (find_package)
-- Configuring incomplete, errors occurred!
```

**Expected behavior**
Install successfully.

**Failure logs** 
[config-x86-windows-out.log](https://github.com/microsoft/vcpkg/files/3742665/config-x86-windows-out.log)

**Comment by cenit** [comment]
I can't reproduce your error. Are they coming from your PR?

```
# Import target "opencv_core" for configuration "Debug"
set_property(TARGET opencv_core APPEND PROPERTY IMPORTED_CONFIGURATIONS DEBUG)
set_target_properties(opencv_core PROPERTIES
  IMPORTED_IMPLIB_DEBUG "${_IMPORT_PREFIX}/debug/lib/opencv_cored.lib"
  IMPORTED_LOCATION_DEBUG "${_IMPORT_PREFIX}/debug/bin/opencv_cored.dll"
  )
```

**Comment by LilyWangL** [comment]
I try to delete vcpkg_fixup_cmake_targets and install opencv4. The *.lib path and *d.dll path in OpenCVModules-debug.cmake both have no debug/.

**Comment by cenit** [comment]
why did you delete the vcpkg_fixup_cmake_targets call? It's necessary of course to fix the paths, and it is working correctly in current OpenCV portfile.

**Comment by LilyWangL** [comment]
Sorry, I didn’t express it clearly. I just want to test if this error is caused by vcpkg_fixup_cmake_targets.
```
CMake Error at F:/Lily/issue8393/vcpkg/installed/x86-windows/share/opencv/OpenCVModules.cmake:164 (message):
  The imported target "opencv_core" references the file

     "F:/Lily/issue8393/vcpkg/installed/x86-windows/x86/vc16/bin/opencv_cored.dll"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "F:/Lily/issue8393/vcpkg/installed/x86-windows/share/opencv/OpenCVModules.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  F:/Lily/issue8393/vcpkg/installed/x86-windows/share/opencv/OpenCVConfig.cmake:126 (include)
  F:/Lily/issue8393/vcpkg/scripts/buildsystems/vcpkg.cmake:263 (_find_package)
  CMakeLists.txt:36 (find_package)
-- Configuring incomplete, errors occurred!
```
If opencv 4.1.2 is installed normally, some ports depend on opencv4 will build failed and get the above error message . I found that the *d.dll path in file OpenCVModules-debug.cmake are missing /debug. Like this:
```
set_target_properties(opencv_core PROPERTIES
  IMPORTED_IMPLIB_DEBUG "${_IMPORT_PREFIX}/debug/lib/opencv_cored.lib"
  IMPORTED_LOCATION_DEBUG "${_IMPORT_PREFIX}/x86/vc16/bin/opencv_cored.dll"
  )
```
If I deleted vcpkg_fixup_cmake_targets and installed opencv4, I found the *.lib path and *d.dll path in OpenCVModules-debug.cmake both have no debug/. So I get this conclusion.

**Comment by cenit** [comment]
That's because you patched badly opencv 4.1.2 I suppose. Maybe these lines are involved? 
https://github.com/microsoft/vcpkg/pull/8557/files#diff-f7f53558f5232928646d66c43cfcf17eR54-R56

**Comment by LilyWangL** [comment]
These lines don't involve it. I have been test this.
> That's because you patched badly opencv 4.1.2 I suppose. Maybe these lines are involved?
> https://github.com/microsoft/vcpkg/pull/8557/files#diff-f7f53558f5232928646d66c43cfcf17eR54-R56
